stix2/equivalence/object/__init__.py

```diff
@@ -439,6 +498,15 @@ def list_reference_check(refs1, refs2, ds1, ds2, **weights):
     return result
 
 
+def _datastore_check(ds1, ds2):
+    if (
+        issubclass(ds1.__class__, (DataStoreMixin, DataSink, DataSource)) or
+        issubclass(ds2.__class__, (DataStoreMixin, DataSink, DataSource))
```

Comment:
I don't think DataSink should be included because you can't `.get()` from a DataSink. Sinks are designed just for adding, not querying.

Reply:
removed DataSink in this check